charts/IndicatorStore.ts

```diff
@@ -0,0 +1,59 @@
+import { fetchJSON } from "./Util"
+import { Indicator } from "./Indicator"
+import { BAKED_BASE_URL } from "settings"
+
+export class IndicatorStore {
+    private indicatorsById: { [id: number]: Indicator } = {}
+    private fetchPromise?: Promise<void>
+
+    private async fetchAll() {
+        try {
+            const json = await fetchJSON(
+                `${BAKED_BASE_URL}/explore/indicators.json`
+            )
+            const indicators: Indicator[] = json.indicators
+            return indicators
+        } catch (error) {
+            console.error("Failed to fetch indicators", error)
+            return []
+        }
+    }
+
+    private loadAllIdempotent() {
+        if (!this.fetchPromise) {
+            this.fetchPromise = new Promise(async resolve => {
+                this.populate(await this.fetchAll())
+                resolve()
+            })
+        }
+        return this.fetchPromise
+    }
+
+    private populate(indicators: Indicator[]) {
+        for (const indicator of indicators) {
+            this.indicatorsById[indicator.id] = indicator
+        }
```

Comment:
Isn't there a util for this? `byId` or `byKey` or something

Reply:
Yeah implemented the equivalent with lodash now